src/core/sagas/autocomplete.js

```diff
@@ -0,0 +1,40 @@
+// Disabled because of
+// https://github.com/benmosher/eslint-plugin-import/issues/793
+/* eslint-disable import/order */
+import { call, put, select, takeLatest } from 'redux-saga/effects';
+/* eslint-enable import/order */
+
+import { AUTOCOMPLETE_STARTED, autocompleteLoad } from 'core/reducers/autocomplete';
+import { autocomplete as autocompleteApi } from 'core/api';
+import log from 'core/logger';
+import { createErrorHandler, getState } from 'core/sagas/utils';
+
+
+export function* fetchAutocompleteResults({ payload }) {
+  const { errorHandlerId } = payload;
+  const errorHandler = createErrorHandler(errorHandlerId);
+
+  yield put(errorHandler.createClearingAction());
+
+  try {
+    const { query } = payload;
+
+    const state = yield select(getState);
+
+    const response = yield call(autocompleteApi, {
+      api: state.api,
+      auth: state.auth,
+      query,
+    });
+    const { results } = response;
+
+    yield put(autocompleteLoad({ results }));
+  } catch (error) {
+    log.warn(`Autcomplete results failed to load: ${error}`);
+    yield put(errorHandler.createErrorAction(error));
+  }
+}
+
+export default function* autocompleteSaga() {
+  yield takeLatest(AUTOCOMPLETE_STARTED, fetchAutocompleteResults);
```

Comment:
You may need to debounce this (there's a [recipe](https://redux-saga.js.org/docs/recipes/) for it) but that can be tweaked later too.